frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -1252,18 +1252,30 @@ func dispatchCopy(d *dispatchState, cfg copyConfig) error {
 				a = a.Copy(st, f, dest, opts...)
 			}
 		} else {
+			var patterns []string
+			if cfg.parents {
+				// detect optional pivot point
+				parent, pattern, ok := strings.Cut(src, "/./")
+				if !ok {
+					pattern = src
+					src = "/"
```

Comment:
os.PathSeparator should be used ?

Reply:
No, it should always be slash. Windows works with slashes as well. If the input is not `/` ( and windows) then it is converted to `/` in `NormalizePath()`.